AdsiPS/Public/Test-ADSICredential.ps1

```diff
@@ -40,20 +48,32 @@ function Test-ADSICredential
 		[string]$AccountName,
 		
 		[Parameter(Mandatory)]
-		[System.Security.SecureString]$AccountPassword
-	)
+		[System.Security.SecureString]$AccountPassword,
+
+		[Alias("RunAs")]
+		[System.Management.Automation.PSCredential]
+        [System.Management.Automation.Credential()]
+		$Credential = [System.Management.Automation.PSCredential]::Empty,
+
+		[String]$DomainName)
 	BEGIN
 	{
 		Add-Type -AssemblyName System.DirectoryServices.AccountManagement
+
+		# Create Context splatting
+		$ContextSplatting = @{ ContextType = "Domain" }
+
+		IF ($PSBoundParameters['Credential']) { $ContextSplatting.Credential = $Credential }
+		IF ($PSBoundParameters['DomainName']) { $ContextSplatting.DomainName = $DomainName }
+
+		$Context = New-ADSIPrincipalContext @ContextSplatting
 	}
 	PROCESS
 	{
 		TRY
 		{
-			$DomainPrincipalContext = New-Object System.DirectoryServices.AccountManagement.PrincipalContext('domain')
-			
-			Write-Verbose -Message "[Test-ADCredential][PROCESS] Validating $AccountName Credential against $($DomainPrincipalContext.ConnectedServer)"
-			$DomainPrincipalContext.ValidateCredentials($AccountName, $AccountPassword)
+			Write-Verbose -Message "[Test-ADSICredential][PROCESS] Validating $AccountName Credential against $($Context.ConnectedServer)"
+			$Context.ValidateCredentials($AccountName, (New-Object PSCredential "user",$AccountPassword).GetNetworkCredential().Password)
```

Comment:
Kind of the same comment, is there a reason why [System.Security.SecureString]$AccountPassword would not do the job ? Is there an advantage to create a new PSCredential object ?

Reply:
Same with this one, [ValidateCredentials](https://msdn.microsoft.com/en-us/library/bb300969.aspx) only takes a password in plaintext. It was testing to see if the password was set to "System.Security.SecureString" every time.